src/auth/AxiosJwtAuthService.js

```diff
@@ -75,6 +75,30 @@ class AxiosJwtAuthService {
       });
   }
 
+  /**
+   *
+   * @param {HttpClient} client An axios instance.
+   * @param {Array|Function} middlewares The middleware to be applied.
+   *
+   */
+  applyMiddleware(client, middlewares = []) {
+    try {
+      (middlewares).forEach((middleware) => {
+        if (typeof middleware === 'function') {
+          middleware(client);
+        } else if (Array.isArray(middleware)) {
+          const [middlewareToApply, options] = middleware;
+          middlewareToApply(client, options);
```

Comment:
I think we get too much benefit from validation here since it could be a bad function or options could be the wrong type. Ideally It should be obvious enough to the consumer if an exception is thrown, or if we have types.